cursorless.nvim/lua/cursorless/cursorless.lua

```diff
@@ -0,0 +1,141 @@
+end
+
+-- https://www.reddit.com/r/neovim/comments/p4u4zy/how_to_pass_visual_selection_range_to_lua_function/
+-- https://neovim.io/doc/user/api.html#nvim_win_get_cursor()
+--
+-- luacheck:ignore 631
+-- e.g. run in command mode :vmap <c-a> <Cmd>lua print(vim.inspect(require('talon.cursorless').buffer_get_selection()))<Cr>
+-- then go in visual mode with "v" and select "hello" on the first line and continue selection with "air"
+-- on the second line.
+-- Then hit ctrl+b and it will show the selection
+-- cline=2, ccol=2, vline=1, vcol=1
+-- sline=1, scol=1, eline=2, ecol=3, reverse=false
+-- { 1, 1, 2, 3, false }
+--
+-- if instead you select from the end of the "air" word on the second line
+-- and select up to the beginning of "hello" on the first line
+-- cline=1, ccol=0, vline=3, vcol=3
+-- sline=1, scol=1, eline=2, ecol=3, reverse=true
+-- { 1, 1, 2, 3, true }
+--
+-- if you want to directly see how it is parsed in the node extension, you can use the below:
+-- e.g. run in command mode :vmap <c-a> <Cmd>:call CursorlessLoadExtension()<Cr>
+-- and again use ctrl+a after selecting the text
+function M.buffer_get_selection()
+  -- print('buffer_get_selection()')
+  local modeInfo = vim.api.nvim_get_mode()
+  local mode = modeInfo.mode
+
+  local cursor = vim.api.nvim_win_get_cursor(0)
+  local cline, ccol = cursor[1], cursor[2]
+  local vline, vcol = vim.fn.line("v"), vim.fn.col("v")
+  -- print(('cline=%d, ccol=%d, vline=%d, vcol=%d'):format(cline, ccol, vcol, vcol))
+
+  local sline, scol
+  local eline, ecol
+  local reverse
+  if cline == vline then
+    --   if ccol <= vcol then
+    if ccol < vcol then
+      sline, scol = cline, ccol
+      eline, ecol = vline, vcol
+      scol = scol + 1
+      reverse = true
+    else
+      sline, scol = vline, vcol
+      eline, ecol = cline, ccol
+      ecol = ecol + 1
+      reverse = false
+    end
+  elseif cline < vline then
+    sline, scol = cline, ccol
+    eline, ecol = vline, vcol
+    scol = scol + 1
+    reverse = true
+  else
+    sline, scol = vline, vcol
+    eline, ecol = cline, ccol
+    ecol = ecol + 1
+    reverse = false
+  end
+
+  if mode == "V" or mode == "CTRL-V" or mode == "\22" then
+    scol = 1
+    ecol = nil
+  end
+
+  -- print(
+  --   ('sline=%d, scol=%d, eline=%d, ecol=%d, reverse=%s'):format(
+  --     sline,
+  --     scol,
+  --     eline,
+  --     ecol,
+  --     tostring(reverse)
+  --   )
+  -- )
+  return { sline, scol, eline, ecol, reverse }
+end
+
+-- https://www.reddit.com/r/neovim/comments/p4u4zy/how_to_pass_visual_selection_range_to_lua_function/
+-- luacheck:ignore 631
+-- e.g. run in command mode :vmap <c-b> <Cmd>lua print(vim.inspect(require('talon.cursorless').buffer_get_selection_text()))<Cr>
+-- then go in visual mode with "v" and select "hello" on the first line and continue selection with "air"
+-- on the second line.
+-- Then hit ctrl+b and it will show the selection
+-- { "hello", "air" }
+function M.buffer_get_selection_text()
```

Comment:
I'm cleaning up a little bit of code around here, and noticed this is using require("talon.cursorless") which isn't valid anymore, so I took a look and it seems like nothing uses this function anywhere? It will be good for you to confirm, but if you don't think we'll use this function I guess can delete it. If it's just for debugging, it's probably worth noting that's why it exists.

Reply:
Iirc i started implementing by "buffer_get_selection_text() " which returns the selected text. Then i realised cursorless expects the indexes so i wrote "buffer_get_selection().

I think we should be able to delete "buffer_get_selection_text()" as i don't think cursorless

Reply:
Maybe @pokey can confirm?